pkg/cloud/common/sources.go

```diff
@@ -0,0 +1,36 @@
+package common
+
+import (
+	"bytes"
+	"embed"
+
+	"k8s.io/apimachinery/pkg/util/yaml"
+	"k8s.io/klog"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+)
+
+type ObjectSource struct {
+	Object client.Object
+	Path   string
+}
+
+func ReadResources(f embed.FS, sources []ObjectSource) ([]client.Object, error) {
+	ret := []client.Object{}
+	for _, source := range sources {
+		data, err := f.ReadFile(source.Path)
+		if err != nil {
+			klog.Errorf("Cannot read embedded resource %v: %v", source.Path, err)
+			return nil, err
+		}
+
+		dec := yaml.NewYAMLOrJSONDecoder(bytes.NewReader(data), 1000)
+		if err := dec.Decode(source.Object); err != nil {
```

Comment:
Should we be copying the source object, otherwise this modifies the input to the function which could be a nasty future side effect. The alternative is to make this a `ReadResourcesInto` style method so it's expected that it reads into the `sources` input